### PR TITLE
refactor!: drop node v12 and eslint v7

### PR DESCRIPTION
**Pull request by amareshsm**
Drop node 12 and upgraded dependencies.

This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
Drop node 12 and upgraded dependencies.

### Breaking Changes
yes, removed node v12 support

### CODE REVIEW

**Comment by amareshsm** [comment]
Bump

**Comment by amareshsm** [comment]
Bumping for review.

File: babel.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ module.exports = (api) => {
         '@babel/preset-env',
         {
           targets: {
-            node: '12.13.0',
+            node: '14.17.0',
```
Comment (ricardogobbosouza) [inline babel.config.js:13]: The default is `14.15.0`

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
   "main": "dist/cjs.js",
   "types": "declarations/index.d.ts",
   "engines": {
-    "node": ">= 12.13.0"
+    "node": ">= 14.17.0"
```
Comment (ricardogobbosouza) [inline package.json:17]: The default is `14.15.0`

File: package.json [hunk 3]
```diff
@@ -47,40 +47,40 @@
   },
   "dependencies": {
     "@types/eslint": "^7.28.2 || ^8.2.0",
```
Comment (ricardogobbosouza) [inline package.json:49]: With this change in node version also need to remove eslint 7 support

**Review by ricardogobbosouza — CHANGES_REQUESTED** [review]
Could you change the `README `?
Its v8 now

```
You also need to install eslint >= 7 from npm, if you haven't already:
```

**Comment by amareshsm** [comment]
Bumping for review.

**Pull request merged**